addons/mail/models/mail_message.py

```diff
@@ -90,9 +90,21 @@ def default_get(self, fields):
     preview = fields.Char(
         'Preview', compute='_compute_preview',
         help='The text-only beginning of the body used as email preview.')
+    link_preview_message_ids = fields.One2many(
+        "mail.link.preview.message",
+        "message_id", domain=[("is_hidden", "=", False)], groups="base.group_erp_manager")
     link_preview_ids = fields.One2many(
-        'mail.link.preview', 'message_id', string='Link Previews',
-        groups="base.group_erp_manager")
+        'mail.link.preview',
+        string='Link Previews',
+        compute="_compute_link_preview_ids",
+        groups="base.group_erp_manager"
+    )
+
+    @api.depends("link_preview_message_ids")
+    def _compute_link_preview_ids(self):
```

Comment:
Would say to try to define ``link_preview_ids`` as a Many2many and remove custom computation. We did that kind of "model hack" in other modules (like slide channel members for example), it helps understanding the model.

Reply:
I tested the many2many with a "custom"  intermediate table, and it didn't work well.
I'll try again since you are asking for it

Reply:
It look like it works this time, so it's indeed better that way :)

Reply:
Yes it is a bit "border line", but it allows to have a m2m like behavior ^^ .

Edit: seems to still be a one2many ?